authority/authority.go

```diff
@@ -338,6 +342,55 @@ func (a *Authority) init() error {
 		},
 		GetIdentityFunc: a.getIdentityFunc,
 	}
+
+	// Check if a KMS with decryption capability is required and available
+	if a.requiresDecrypter() {
+		if _, ok := a.keyManager.(kmsapi.Decrypter); !ok {
+			return errors.New("keymanager doesn't provide crypto.Decrypter")
+		}
+	}
+
+	// TODO: decide if this is a good approach for providing the SCEP functionality
+	// It currently mirrors the logic for the x509CAService
+	if a.requiresSCEPService() && a.scepService == nil {
+		var options casapi.Options
+		if a.config.AuthorityConfig.Options != nil {
+			options = *a.config.AuthorityConfig.Options
+		}
+
+		// Read intermediate and create X509 signer and decrypter for default CAS.
+		if options.Is(casapi.SoftCAS) {
+			options.CertificateChain, err = pemutil.ReadCertificateBundle(a.config.IntermediateCert)
+			if err != nil {
+				return err
+			}
+			options.Signer, err = a.keyManager.CreateSigner(&kmsapi.CreateSignerRequest{
+				SigningKey: a.config.IntermediateKey,
+				Password:   []byte(a.config.Password),
+			})
+			if err != nil {
+				return err
+			}
+
+			if km, ok := a.keyManager.(kmsapi.Decrypter); ok {
+				options.Decrypter, err = km.CreateDecrypter(&kmsapi.CreateDecrypterRequest{
+					DecryptionKey: a.config.IntermediateKey,
```

Comment:
Right now SCEP indeed requires the intermediate key for decryption. A SCEP client uses a CA certificate, in this case the intermediate certificate, to encrypt messages. The SCEP authority needs the key to decrypt those messages.

The CA certificate does not need to be the same as the intermediate; it can be a different certificate/key combination. SCEP is normally performed over HTTP, meaning there's no reason to have an HTTPS connection with corresponding CA. The current configuration is kind of a convenient way to run SCEP. I think it may even be good idea to totally separate the "TLS CA" from the "SCEP CA", but I'm not an expert in this area.

I'm not sure if the key should be in the provisioner directly. I do think it's a good thing to make it possible to control the key that's used by the SCEP authority via the provisioner. The actual key should IMO exist in a KMS, but I think we agree on that part already.